ts_src/payments/p2air.ts

```diff
@@ -0,0 +1,178 @@
+export function p2air(a: Payment, opts?: PaymentOpts): Payment {
+  if (!a.input && !a.output && !a.pubkeys && !a.signatures)
+    throw new TypeError('Not enough data');
+  if (a.vaultTxType === undefined) a.vaultTxType = VaultTxType.Alert;
+
+  opts = Object.assign({ validate: true }, opts || {});
+
+  function isAcceptableSignature(x: Buffer | number): boolean {
+    return (
+      bscript.isCanonicalScriptSignature(x as Buffer) ||
+      (opts!.allowIncomplete && (x as number) === OPS.OP_0) !== undefined
+    );
+  }
+
+  typef(
+    {
+      network: typef.maybe(typef.Object),
+      output: typef.maybe(typef.Buffer),
+      pubkeys: typef.maybe(typef.arrayOf(ecc.isPoint)),
+      signatures: typef.maybe(typef.arrayOf(isAcceptableSignature)),
+      input: typef.maybe(typef.Buffer),
+      vaultTxType: typef.maybe(typef.Number),
+    },
+    a,
+  );
+
+  const network = a.network || BITCOIN_NETWORK;
+  const o: Payment = { network };
+
+  let chunks: Stack = [];
+  let decoded = false;
+  function decode(output: Buffer | Stack): void {
+    if (decoded) return;
+    decoded = true;
+    chunks = bscript.decompile(output) as Stack;
+    o.pubkeys = chunks.slice(9, 12) as Buffer[];
+  }
+
+  lazy.prop(o, 'output', () => {
+    if (!a.pubkeys) return;
+    return bscript.compile(
+      ([] as Stack).concat(
+        OPS.OP_IF,
+        OP_INT_BASE + 1,
+        OPS.OP_ELSE,
+        OPS.OP_IF,
+        OP_INT_BASE + 2,
+        OPS.OP_ELSE,
+        OP_INT_BASE + 3,
+        OPS.OP_ENDIF,
+        OPS.OP_ENDIF,
+        a.pubkeys,
+        OP_INT_BASE + 3,
+        OPS.OP_CHECKMULTISIG,
+      ),
+    );
+  });
+  lazy.prop(o, 'm', () => {
+    switch (o.vaultTxType) {
+      case VaultTxType.Alert:
+        return 1;
+      case VaultTxType.Instant:
+        return 2;
+      case VaultTxType.Recovery:
+      default:
+        return 3;
+    }
+  });
+  lazy.prop(o, 'n', () => {
+    return 3;
+  });
+  lazy.prop(o, 'pubkeys', () => {
+    if (!a.output) return;
+    decode(a.output);
+    return o.pubkeys;
+  });
+  lazy.prop(o, 'signatures', () => {
+    if (!a.input) return;
+    const lastSignaturePos = a.vaultTxType === VaultTxType.Alert ? -1 : -2;
+    return bscript.decompile(a.input)!.slice(1, lastSignaturePos);
+  });
+  lazy.prop(o, 'input', () => {
+    if (!a.signatures) return;
+    let flags;
+    switch (o.vaultTxType) {
+      case VaultTxType.Alert:
+        flags = [OPS.OP_1];
+        break;
+      case VaultTxType.Instant:
+        flags = [OPS.OP_1, OPS.OP_0];
+        break;
+      case VaultTxType.Recovery:
+      default:
+        flags = [OPS.OP_0, OPS.OP_0];
+    }
+    return bscript.compile(
+      ([OPS.OP_0] as Stack).concat(a.signatures).concat(flags as Stack),
+    );
+  });
+  lazy.prop(o, 'witness', () => {
+    if (!o.input) return;
+    return [];
+  });
+  lazy.prop(o, 'name', () => {
+    return `p2air(${VaultTxType[o.vaultTxType as VaultTxType]})`;
```

Comment:
please try to remove every place when "as" is used and write code that handles it. The typescript is trying to help you so the code has no bugs